### PR TITLE
Avoid redundant statement branching in safeToSecureCookie()

### PR DESCRIPTION
**Pull request by cschanaj**
#17711

### CODE REVIEW

**Comment by pipboy96** [comment]
@cschanaj simply changing `var` to `let` should probably be done as a separate PR.

File: chromium/background-scripts/rules.js [hunk 1]
```diff
@@ -614,7 +614,7 @@ RuleSets.prototype = {
    * @returns {*} ruleset or null
    */
   shouldSecureCookie: function(cookie) {
-    var hostname = cookie.domain;
+    let hostname = cookie.domain;
```
Comment (pipboy96) [inline chromium/background-scripts/rules.js]: Keep as `var`, this should be done as part of ESLint.

File: chromium/background-scripts/rules.js [hunk 2]
```diff
@@ -624,12 +624,11 @@ RuleSets.prototype = {
       return null;
     }
 
-    var potentiallyApplicable = this.potentiallyApplicableRulesets(hostname);
+    let potentiallyApplicable = this.potentiallyApplicableRulesets(hostname);
```
Comment (pipboy96) [inline chromium/background-scripts/rules.js]: Keep as `var`, this should be done as part of ESLint.

File: chromium/background-scripts/rules.js [hunk 3]
```diff
@@ -627,9 +627,8 @@ RuleSets.prototype = {
     var potentiallyApplicable = this.potentiallyApplicableRulesets(hostname);
     for (let ruleset of potentiallyApplicable) {
       if (ruleset.cookierules !== null && ruleset.active) {
-        for (let cookierules of ruleset.cookierules) {
-          var cr = cookierules;
-          if (cr.host_c.test(cookie.domain) && cr.name_c.test(cookie.name)) {
+        for (let cookierule of ruleset.cookierules) {
```
Comment (Hainish) [inline chromium/background-scripts/rules.js]: `const` can be used here

**Pull request merged**